presto-native-execution/presto_cpp/main/CMakeLists.txt

```diff
@@ -121,3 +121,10 @@ if(PRESTO_STATS_REPORTER_TYPE)
         "${PRESTO_STATS_REPORTER_TYPE} is not a valid stats reporter name")
   endif()
 endif()
+
+# Conditionally add Linux-specific source files. Add as a source to
+# presto_server to allow global reporter Singleton initialization.
+if(UNIX AND NOT APPLE)
```

Comment:
You can probably add `if(PRESTO_MEMORY_CHECKER_TYPE STREQUAL "LINUX_MEMORY_CHECKER")` here and avoid the `add_compile_definitions` suggestion below.